### PR TITLE
Feat/aag 2819 additional options on load support

### PR DESCRIPTION
**Pull request by thomas-james-1986**
## JIRA
[AAG-2819]

## DESCRIPTION

This PR adds the Objective-C changes to support sending additional parameters on load for iOS

## SCREENSHOTS
n/a

[AAG-2819]: https://superawesomeltd.atlassian.net/browse/AAG-2819?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Pod/Plugin/Unity/SAUnityBannerAd.m [hunk 1]
```diff
@@ -53,8 +53,15 @@ void SuperAwesomeUnitySABannerAdCreate (const char *unityName) {
  * @param placementId   placement id to load the ad for
  * @param configuration production = 0 / staging = 1 --- deprecated
  * @param test          true / false
+ * @param encodedOptions a json encoded dictionary of options to send with requests
  */
-void SuperAwesomeUnitySABannerAdLoad (const char *unityName, int placementId, int configuration, bool test) {
+void SuperAwesomeUnitySABannerAdLoad(const char *unityName, int placementId, int configuration, bool test, const char *encodedOptions) {
+
+    NSString *options = [NSString stringWithUTF8String:encodedOptions];
```
Comment (gunhansancar) [inline Pod/Plugin/Unity/SAUnityBannerAd.m:60]: I wonder if we can check that `encodedOptions` is null before trying to decode? This applies the similar methods below.

Comment (thomas-james-1986) [reply Pod/Plugin/Unity/SAUnityBannerAd.m:60]: It's a good point, in Objective-C `[options length]` resolves to 0 if options is null though so it counts as a null check. I will re-test this though to be sure it's safe

Comment (thomas-james-1986) [reply Pod/Plugin/Unity/SAUnityBannerAd.m]: Ok nice catch, the line`NSString *options = [NSString stringWithUTF8String:encodedOptions];` could fail if null is passed for `encodedOptions`. It was _safe_ as this is a middle layer between the public C# and the iOS SDK and the C# always passes a default but this is safer 👍. I've re-tested for banners, videos and interstitials with null, the original load method signature and an options dictionary.

Comment (thomas-james-1986) [reply Pod/Plugin/Unity/SAUnityBannerAd.m]: I think we should test the string length on `encodedOptions` too, I'll make the changes

File: Pod/Plugin/Unity/SAUnityBannerAd.m [hunk 2]
```diff
@@ -63,13 +65,27 @@ void SuperAwesomeUnitySABannerAdLoad (const char *unityName, int placementId, in
...
-        
+
         SABannerAd *banner = [bannerDictionary objectForKey:key];
         [banner setTestMode:test];
-        [banner load:placementId];
-        
-    } else {
-        // handle failure
+
+        if(encodedOptions) {
```
Comment (me2k) [inline Pod/Plugin/Unity/SAUnityBannerAd.m:72]:
```suggestion
        if (encodedOptions) {
```

File: Pod/Plugin/Unity/SAUnityVideoAd.m [hunk 3]
```diff
@@ -34,12 +34,33 @@ void SuperAwesomeUnitySAVideoAdCreate () {
...
  */
-void SuperAwesomeUnitySAVideoAdLoad(int placementId, int configuration, bool test, int playback) {
+
+void SuperAwesomeUnitySAVideoAdLoad(int placementId, int configuration, bool test, int playback, const char *encodedOptions) {
+
     [SAVideoAd setTestMode:test];
     [SAVideoAd setPlaybackMode:[StartDelayHelper from:playback]];
-    [SAVideoAd load: placementId];
+
+    if(encodedOptions) {
```
Comment (me2k) [inline Pod/Plugin/Unity/SAUnityVideoAd.m:47]:
```suggestion
    if (encodedOptions) {
```

**Review by me2k — APPROVED** [review]

**Pull request merged**